### PR TITLE
 Limit number of URLs per request #54

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

File: app/service.js [hunk 1]
```diff
@@ -48,6 +51,12 @@ app.post('/v1/metadata', function(req, res) {
     return;
   }
 
+  const maxUrls = app.get('maxUrls');
+  if (req.body.urls.length > maxUrls) {
+    fail(errorMessages.maxUrls, 400);
```
Comment (pdehaan) [inline app/service.js]: I'm a bit confused on the desired behavior.
If I send you 35 URLs (with a max of 20), should it:

a) Fail and give me nothing?
b) Return me the first 20 results and ignore the rest?
c) Return me the first 20 results, and return an error response for the rest?

For example,

```js
  {
    request_error: "",
    url_errors: {
      "http://www.mozilla.org/21": "Fetch Error: maxUrls exceeded"
    },
    urls: {
      "http://www.mozilla.org/1": {},
      ...
      "http://www.mozilla.org/20": {}
    }
  }
```

/cc @k88hudson

Comment (jaredkerim) [inline app/service.js]: The current behaviours of the existing services are to reject the call entirely as invalid if the number of URLs requested exceeds the configured maximum:

https://github.com/mozilla/embedly-proxy/blob/master/embedly/embedly/tests/test_api.py#L103-L112

Comment (jaredkerim) [inline app/service.js]: The onus lives on the caller to break their requested workload into manageable chunks, see here:

https://github.com/mozilla/activity-stream/blob/master/addon/PreviewProvider.js#L256-L263

**Comment by pdehaan** [comment]
Looks kosher. r+

👍

**Pull request merged**